src/chromium/adding-third-party-crates/reviews-and-audits.md

```diff
@@ -0,0 +1,29 @@
+# Auditing third party crates
+
+Adding new libraries is subject to Chromium's standard [policies][0], but of
+course also subject to code review. As you may be bringing in not just a single
+crate but also transitive depedencies, there may be a lot of code to review.
+On the other hand, safe Rust code can have limited negative side effects.
+How should you review it?
+
+Over time Chromium aims to move to a process based around [cargo vet][1].
+
+Meanwhile, for each new crate addition, we are checking for the following:
+
+* Understand why each crate is used. What's the relationship between crates?
+  If the build system for each crate contains a `build.rs` or procedural
+  macros, work out what they're for. Are they compatible with the way
+  Chromium is normally built?
+* Check each crate seems to be reasonably well maintained
+* Use `cargo audit` to check for known vulnerabilities (TODO: how to run this)
+* Ensure any unsafe code is good enough for the [Rule of Two][2]
+* Check for any use of `fs` or `net` APIs
+* Read all the code at a sufficient level to look for anything out of place
+  that might have been maliciously inserted. (You can't realistically aim
+  for 100% perfection here: there's often just too much code.)
+
+These are just guidelines - you should satisfy yourself however you see fit.
```

Comment:
"however you see fit" seems inaccurate.  Maybe: "please work with a reviewer from chromium-security@ to figure out what to look for".

Reply:
Done